lib/content/message/predictions.ex

```diff
@@ -154,17 +196,101 @@ defmodule Content.Message.Predictions do
+  defp do_crowding(prediction) when prediction.route_id in ["Orange"] do
+    case Engine.Locations.for_vehicle(prediction.vehicle_id) do
+      %Locations.Location{} = location ->
+        {calculate_crowding_data_confidence(prediction, location),
+         get_crowding_description(location.multi_carriage_details)}
+
+      _ ->
+        {nil, nil}
+    end
+  end
+
+  defp do_crowding(_), do: {nil, nil}
 
   defp calculate_crowding_data_confidence(prediction, location)
-       when prediction.route_id in ["Orange"] and location.stop_id == prediction.stop_id do
+       when location.stop_id == prediction.stop_id do
     if location.status in [:incoming_at, :in_transit_to],
       do: :high,
       else: :low
   end
 
   defp calculate_crowding_data_confidence(_prediction, _location), do: nil
 
+  defp get_crowding_description([]), do: nil
+
+  defp get_crowding_description(carriage_details) do
+    crowding_level_by_car =
+      Enum.map(carriage_details, fn carriage ->
+        {carriage.carriage_sequence,
+         occupancy_status_to_crowding_level(carriage.occupancy_status)}
+      end)
+
+    min_crowding_level = Enum.min_by(crowding_level_by_car, &elem(&1, 1)) |> elem(1)
+
+    emptier_car_nums =
+      Stream.map(crowding_level_by_car, fn {car_num, crowding_level} ->
+        if crowding_level > min_crowding_level,
+          do: {car_num, :fuller},
+          else: {car_num, :emptier}
+      end)
+      |> Stream.filter(&match?({_, :emptier}, &1))
+      |> Enum.map(&elem(&1, 0))
+
+    emptier_location =
+      Enum.reduce_while(
+        @crowding_zone_definitions[Enum.count(emptier_car_nums)],
+        :front_and_back,
+        fn {zone, definition}, acc ->
+          if in_zone?(emptier_car_nums, zone, definition),
+            do: {:halt, zone},
+            else: {:cont, acc}
+        end
+      )
+
+    {emptier_location, crowding_level_to_atom(min_crowding_level)}
+  end
+
+  defp occupancy_status_to_crowding_level(occupancy_status) do
+    case occupancy_status do
+      :many_seats_available -> @not_crowded
+      :few_seats_available -> @not_crowded
+      :standing_room_only -> @some_crowding
+      :crushed_standing_room_only -> @crowded
+      :full -> @crowded
+      _ -> @unknown_crowding
+    end
+  end
+
+  defp crowding_level_to_atom(crowding_level) do
+    case crowding_level do
+      @not_crowded -> :not_crowded
+      @some_crowding -> :some_crowding
+      @crowded -> :crowded
+      @unknown_crowding -> :unknown_croding
+    end
+  end
+
+  defp in_zone?(_, _, nil), do: false
+
+  defp in_zone?(car_nums, zone, {front_end, back_end}) do
+    special_case? =
+      (Enum.count(car_nums) == 2 and zone == :front and car_nums == [1, 4]) or
+        (Enum.count(car_nums) == 2 and zone == :back and car_nums == [3, 6]) or
+        (Enum.count(car_nums) == 4 and zone == :front and car_nums == [1, 2, 3, 6]) or
+        (Enum.count(car_nums) == 4 and zone == :back and car_nums == [1, 4, 5, 6])
```

Comment:
Since we're doing a full equality test on `car_nums`, it looks like the `count` checks are redundant